qiskit_experiments/library/randomized_benchmarking/mirror_rb_analysis.py

```diff
@@ -0,0 +1,674 @@
+                    outcomes.append(
+                        AnalysisResultData(
+                            name=f"EPG_{gate}",
+                            value=epg_val,
+                            chisq=fit_data.reduced_chisq,
+                            quality=quality,
+                            extra=metadata,
+                        )
+                    )
+
+        return outcomes
+
+    def _run_data_processing(
+        self, raw_data: List[Dict], models: List[lmfit.Model]
+    ) -> curve.CurveData:
+        """Manual data processing
+
+        Args:
+            raw_data: Payload in the experiment data.
+            models: A list of LMFIT models that provide the model name and
+                optionally data sorting keys.
+
+        Returns:
+            Processed data that will be sent to the formatter method.
+
+        Raises:
+            DataProcessorError: When model is multi-objective function but
+                data sorting option is not provided.
+            DataProcessorError: When key for x values is not found in the metadata.
+        """
+        x_key = self.options.x_key
+
+        try:
+            xdata = np.asarray([datum["metadata"][x_key] for datum in raw_data], dtype=float)
+        except KeyError as ex:
+            raise DataProcessorError(
+                f"X value key {x_key} is not defined in circuit metadata."
+            ) from ex
+
+        ydata = self._compute_polarizations_and_probabilities(raw_data)
+        shots = np.asarray([datum.get("shots", np.nan) for datum in raw_data])
+
+        def _matched(metadata, **filters):
+            try:
+                return all(metadata[key] == val for key, val in filters.items())
+            except KeyError:
+                return False
+
+        if len(models) == 1:
+            # all data belongs to the single model
+            data_allocation = np.full(xdata.size, 0, dtype=int)
+        else:
+            data_allocation = np.full(xdata.size, -1, dtype=int)
+            for idx, sub_model in enumerate(models):
+                try:
+                    tags = sub_model.opts["data_sort_key"]
+                except KeyError as ex:
+                    raise DataProcessorError(
+                        f"Data sort options for model {sub_model.name} is not defined."
+                    ) from ex
+                if tags is None:
+                    continue
+                matched_inds = np.asarray(
+                    [_matched(d["metadata"], **tags) for d in raw_data], dtype=bool
+                )
+                data_allocation[matched_inds] = idx
+
+        return curve.CurveData(
+            x=xdata,
+            y=unp.nominal_values(ydata),
+            y_err=unp.std_devs(ydata),
+            shots=shots,
+            data_allocation=data_allocation,
+            labels=[sub_model._name for sub_model in models],
+        )
+
+    def _compute_polarizations_and_probabilities(self, raw_data: List[Dict]) -> unp.uarray:
```

Comment:
This must be data processor. You can instantiate data processor on the fly with analysis option (`y_axis`), and such mechanism can be implemented `_initialize` method. I would implement three independent nodes for different quantities and just choose a proper node from CurveAnalysis to decouple `y_axis` option from the core data processing logic.

Reply:
I've refactored this as a data processor node, but I thought it was better to keep it as one node with an `analyzed_quantity` parameter instead of three since there was so much shared code. What do you think?

Reply:
It depends how you want to use this processor. If this kind of data processing can be used for other experiment the node should be decoupled from the context of MRB (because I don't think `analyzed_quantity` appears in other analysis). If this is dedicated, probably current implementation is fine. In that case I would make it protected class so that other experiment cannot use it.

Reply:
I think it's fairly specialized to this experiment so I will make it a protected class.